### PR TITLE
Support file-level resource definition annotation

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: Google.Api.Generator.Tests/ProtoTests/CommonResource/CommonResource.proto [hunk 1]
```diff
@@ -15,15 +15,11 @@ service CommonResource {
...
-message Project {
-  option (google.api.resource) = {
+// Also defined as a common resource in CommonResourceConfig.json
+option (google.api.resource_definition) = {
     type: "commonresource.example.com/Project",
 	pattern: "projects/{project}"
-  };
-  string name = 1;
-}
+};
```
Comment (jskeet) [inline Google.Api.Generator.Tests/ProtoTests/CommonResource/CommonResource.proto]: Is there a reason why pattern is indented further than type here?

File: Google.Api.Generator/ProtoUtils/ResourceDetails.cs [hunk 2]
```diff
@@ -156,13 +156,17 @@ public static IReadOnlyList<Definition> LoadResourceDefinitionsByFileName(IEnume
             // Compatibility problems: Adding a new resource with the same pattern as a single-pattern parent.
             //                         Adding a new resource with the same pattern as one of a multi-pattern parent.
             // TODO: Support new (Sept 2019) `name_descriptor` way of specifying resource-names.
-            var msgs = descs
+            var msgsFromProtoMsgs = descs
```
Comment (jskeet) [inline Google.Api.Generator/ProtoUtils/ResourceDetails.cs]: Is the TODO here still valid?

File: Google.Api.Generator.Tests/ProtoTests/CommonResource/CommonResource.proto (hunk 1 above)
Comment (chrisdunelm) [reply Google.Api.Generator.Tests/ProtoTests/CommonResource/CommonResource.proto]: Tab/space confusion. Fixed.

File: Google.Api.Generator/ProtoUtils/ResourceDetails.cs (hunk 2 above)
Comment (chrisdunelm) [reply Google.Api.Generator/ProtoUtils/ResourceDetails.cs]: Yes. This is a separate annotation change that's currently still internal-only. I don't know when it's expected to become external.

**Comment by chrisdunelm** [comment]
Thanks. Will merge on green.

**Pull request merged**